File: solutions/python/bob/1/bob.py
import re


def response(hey_bob):
    hey_bob = hey_bob.strip()

    if (hey_bob == ""):
        return "Fine. Be that way!"

    q = hey_bob[-1] == "?"
    s = hey_bob == hey_bob.upper()

    if (re.search("[A-z]", hey_bob)):
        if (s and q):
            return "Calm down, I know what I'm doing!"

        if (s):
            return "Whoa, chill out!"

    if (q):
        return "Sure."

    return "Whatever."

File: solutions/javascript/bob/1/bob.js
export const hey = (message) => {
	message = message.trim();
	let yell = message.toUpperCase() === message,
		question = message.split("").pop() === "?";

	return message === ""
		? "Fine. Be that way!"
		: message.match(/([A-Za-z])/g) === null
		? question
			? "Sure."
			: "Whatever."
		: yell && question
		? "Calm down, I know what I'm doing!"
		: yell
		? "Whoa, chill out!"
		: question
		? "Sure."
		: "Whatever.";
};
